yoconfigurator/dicts.py

```diff
@@ -15,7 +26,7 @@ class DotDict(dict):
 
     def __init__(self, *args, **kwargs):
         super(DotDict, self).__init__(*args, **kwargs)
-        for key, value in self.iteritems():
+        for key, value in iteritems(self):
```

Comment:
I would just use `.items()` here and below. IIRC, .items() does the "right thing" now in Python 3, so using items will be less performant on 2.7 but that doesn't really seem like the end of the world in this case.